File: frontend/src/classes/attachments.ts
export class Attachments {
    attachmentId?: number;
    title?: string;
    cover?: string;
    desc?: string;
    status?: Status;
    privacy?: Privacy;
    isFeatured?: boolean;
    dateCreated?: string | Date;
    dateModified?: string | Date;
    dateDeleted?: string | Date;
    usersId?: number;
}

type Status = "published" | "updated" | "deleted" | "locked";
type Privacy = "public" | "private" | "only_friends_and_followers" | "only_friends" | "only_followers";

File: backend/API/aspnetcore/lcpsnapi/Classes/Attachments.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using static lcpsnapi.Classes.Enums;

namespace lcpsnapi.Classes
{
    public class Attachments
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int? AttachmentId { get; set; }
        public string? Title { get; set; }
        public string? Cover { get; set; }
        public string? Desc { get; set; }
        public Status? Status { get; set; }
        public Privacy? Privacy { get; set; }
        public bool? IsFeatured { get; set; }
        public string? DateCreated { get; set; }
        public string? DateModified { get; set; }
        public string? DateDeleted { get; set; }
        public int? UsersId { get; set; }
    }
}